util/quicserver.c

```diff
@@ -94,23 +95,22 @@ static BIO *create_dgram_bio(int family, const char *hostname, const char *port)
         /* Start listening on the socket */
         if (!BIO_listen(sock, BIO_ADDRINFO_address(ai), 0)) {
             BIO_closesocket(sock);
-            sock = -1;
             continue;
         }
 
         /* Set to non-blocking mode */
         if (!BIO_socket_nbio(sock, 1)) {
             BIO_closesocket(sock);
-            sock = -1;
             continue;
         }
+        bound_socks++;
     }
```

Comment:
This loop doesn't actually quite look right (a pre-existing issue I think). There should be a "break" at the end of this, i.e. if we get to the end of the "for" block and haven't hit "continue", then we have successfully bound a socket and we should stop looking for more sockets. As things stand we loop through all the sockets, create an fd for each one, but then only use the last one.

Reply:
With `break` `ai == NULL` could be used as the failure condition.

Reply:
Agreed, this leaks sockets as-is...

Reply:
I just pushed  a fixup commit to do exactly that.  We now break the loop as soon as we successfully bind and listen on a socket, which mirrors what the clients do on connect